optuna/storages/_rdb/storage.py

```diff
@@ -146,14 +188,14 @@ def __setstate__(self, state: Dict[Any, Any]) -> None:
 
     def create_new_study(self, study_name: Optional[str] = None) -> int:
 
-        session = self.scoped_session()
-
-        if study_name is None:
-            study_name = self._create_unique_study_name(session)
+        try:
+            with _scoped_session(self.scoped_session) as session:
+                if study_name is None:
+                    study_name = self._create_unique_study_name(session)
 
-        study = models.StudyModel(study_name=study_name, direction=StudyDirection.NOT_SET)
-        session.add(study)
-        if not self._commit_with_integrity_check(session):
+                study = models.StudyModel(study_name=study_name, direction=StudyDirection.NOT_SET)
+                session.add(study)
+        except StorageInternalError:
```

Comment:
Should it be `IntegrityError` to keep the behavior unchanged?
```suggestion
        except IntegrityError:
```

Reply:
I agree with you. If I throw `IntegrityError` in the context manager and catch it here, we can keep backward compatibility here. 

I modified the code as 
https://github.com/optuna/optuna/pull/1628/commits/31ac6bfc9c2503f1f703b8c4a40b7330c7ee2c8a .

One concern is that we may throw `IntegrityError` instead of `StorageInternalError` where `_commit` was used. So, users will receive  `IntegrityError` instead of `StorageInternalError`, but it should be minor problem. (Basically, there are no error handlings for both `StorageInternalError` and `OptunaError`, so users will receive errors directly.)